### DISCUSSION TITLE
Port position not working on custom element?

### DISCUSSION THREAD

**Discussion by igo** [clientIO/joint#1845] [Q&A] [answered]
I have a custom `Element` and `ElementView` that does not utilize markup attribute but rather creates everything using `V`:

```ts
class MyElement extends dia.Element {
    ... // no markup in this class
}
class MyElementView extends dia.ElementView {
    render() {
        this.vBody = V('rect').addClass('body');
        this.vel.empty().append([ this.vBody ]);
        ....
    }
}
```

Problem is now with specifying port position. All port positions (left/right/top/bottom) are on the left side as shown on picture. How can I make these positions work?

<img width="293" alt="Screenshot 2022-10-15 at 21 18 21" src="https://user-images.githubusercontent.com/55597/196004316-540c70fa-2147-48d7-80ab-706ac2d7d143.png">

**Comment by kumilingus** [comment, marked as answer]
You might need to call [_renderPorts()](https://github.com/clientIO/joint/blob/master/src/dia/ElementView.mjs#L222) at some point.
Perhaps you didn't set the model's `width`.

It's hard to say without the code. Do you mind creating a JSFiddle?

**Reply by igo** [reply]
Thanks for the info. You solved my other problem when ports were not showing up when loading from JSON. This should be mentioned in the demos/docs.
